### PR TITLE
Add slice op

### PR DESCRIPTION
**Pull request by sfvaroglu**
## Description ##
This PR adds slice op to the bridge.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ops/slice.cc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  ngraph::AxisSet axes;
+  bool reverse = false;
+
+  for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
+    const int len = tshape[i];
+
+    int s = 1;
+    if (param.step[i].has_value()) {
+      s = param.step[i].value();
+      if (s == 0) s = 1;
```
Comment (cconvey) [inline src/ngraph/ops/slice.cc]: Style suggestion: Always use `{` and `}`, even with single-statement bodies.  IME it's an overall productivity win, when considering the bug-hunting it saves.

This comment applies to a few additional lines of code further down.

File: src/ngraph/ops/slice.cc [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+namespace ngraph_bridge {
+
+NgraphNodePtr create_slice_op(const NgraphNodePtr& node,
+                              const nnvm::NodeAttrs& attrs) {
+  const mxnet::op::SliceParam& param =
+      nnvm::get<mxnet::op::SliceParam>(attrs.parsed);
+  nnvm::TShape tshape = NShape_to_TShape(node->get_shape());
+  ngraph::Coordinate ng_begin, ng_end, ng_step;
+  ngraph::AxisSet axes;
+  bool reverse = false;
```
Comment (cconvey) [inline src/ngraph/ops/slice.cc]: Suggestion: IME the algorithm would be slightly clearer if the value of `reverse` was computed just once, early on.  E.g., 

``` c++
...

const bool reverse = (blah blah blah);

for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
   // The 'for' loop as currently written, minus the 'reverse = true;' statement.
   ...
}
```

Comment (sfvaroglu) [reply src/ngraph/ops/slice.cc]: I understand, but `reverse` depends on `step`, which is a tuple. So, we still need to go through the loop.

Comment (cconvey) [reply src/ngraph/ops/slice.cc]: > I understand, but reverse depends on step, which is a tuple. So, we still need to go through the loop.

Right, sorry that my original comment was too hand-wavy to be clear.

I was suggesting that it might be an overall win for readability if we make two passes through the loop: A first pass to determine the value of `reverse`, and then a second pass to do (most of) what you currently do in the loop.

Here's what I meant, in more detail:
``` c++
const bool reverse = std::any_of(param.step.begin(), param.step.end(), 
  [](const dmlc::optional<int> & s) {return s.has_value() && s.value() < 0;});

for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
   ...
}
```

File: src/ngraph/ops/slice.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace ngraph_bridge {
+
+NgraphNodePtr create_slice_op(const NgraphNodePtr& node,
+                              const nnvm::NodeAttrs& attrs) {
+  const mxnet::op::SliceParam& param =
+      nnvm::get<mxnet::op::SliceParam>(attrs.parsed);
+  nnvm::TShape tshape = NShape_to_TShape(node->get_shape());
+  ngraph::Coordinate ng_begin, ng_end, ng_step;
+  ngraph::AxisSet axes;
```
Comment (cconvey) [inline src/ngraph/ops/slice.cc]: It would be nice for `axes` to be named that made its purpose / intent clearer to casual readers of the code.
